src/main/java/by/home/logic/Calculator.java

```diff
@@ -0,0 +1,98 @@
+package by.home.logic;
+
+import by.home.controller.DataReader;
+import by.home.entity.Circle;
+import by.home.entity.Point;
+import org.apache.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Calculator {
+
+    public double calculateArea(Circle circle) {
+        double circleArea = Math.PI * Math.pow(circle.getRadius(), 2);
+        return circleArea;
+    }
+
+    public double calculatePerimetr(Circle circle) {
+        double circlePerimetr = 2 * Math.PI * circle.getRadius();
+        return circlePerimetr;
+    }
+
+    public boolean isCrossingAxisOnDistance(Circle circle, double distance, Axis axis) {
+        List<Point> crossingPointsWithAxis = new ArrayList<Point>();
+        boolean result = false;
+        switch (axis) {
+            case OX:
+                crossingPointsWithAxis = calculatePointsOfCrossingCircleWithOX(circle);
+                if (distance == Math.abs(crossingPointsWithAxis.get(0).getX())
+                        || distance == Math.abs(crossingPointsWithAxis.get(1).getX())) ;
+            {
+                result = false;
+            }
+            break;
+            case OY:
+                crossingPointsWithAxis = calculatePointsOfCrossingCircleWithOY(circle);
+                if (distance == Math.abs(crossingPointsWithAxis.get(0).getY())
+                        || distance == Math.abs(crossingPointsWithAxis.get(1).getY())) ;
+            {
+                result = false;
+            }
+            break;
```

Comment:
Methods are similiar but they have pracrically only "if" and how create new method for me understandable